### PR TITLE
Replace Category pages with search results pages

### LINKED ISSUE #8079: Replace Category pages with search results pages
This is a spin-off from #8050 which captures the decisions made and described at https://github.com/mozilla/addons-frontend/issues/8050#issuecomment-494960622.

We will be doing away with category pages which show multiple shelves for a given category. Instead, any link to a category page will result in a Search Results page being displayed, with the search criteria in the URL specified as being limited to the given category. 

These search results will be sorted by recommended extensions, followed by ADI.

~~When a user chooses a new sort criteria from the form on this search results page, we will ensure that the new search uses recommended extensions as the primary sort criteria in order to force all recommended add-ons to the top.~~ - This has been moved to #8084.

### PR DESCRIPTION
**Pull request by bobsilverberg**
Fixes #8079

### CODE REVIEW

File: src/amo/components/Categories/index.js [hunk 1]
```diff
@@ -146,12 +148,21 @@ export class CategoriesBase extends React.Component<InternalProps> {
               // $FLOW_IGNORE
               const { name, slug } = category;
 
+              const linkTo = {
+                pathname: '/search/',
+                query: convertFiltersToQueryParams({
```
Comment (kumar303) [inline src/amo/components/Categories/index.js]: The results for this new URL seem quite different than the old URL. Is that expected? I was looking at these:

* https://addons.mozilla.org/en-US/firefox/extensions/alerts-updates/
* https://addons.mozilla.org/en-US/firefox/search/?category=alerts-updates&sort=recommended%2Cusers&type=extension

File: src/amo/components/Categories/index.js [hunk 2]
```diff
@@ -146,12 +148,21 @@ export class CategoriesBase extends React.Component<InternalProps> {
...
+                  sort: `${SEARCH_SORT_RECOMMENDED},${SEARCH_SORT_POPULAR}`,
+                }),
+              };
+
               return (
                 <li className="Categories-item" key={name}>
                   <Button
                     className={`Categories-link
                       Categories--category-color-${getCategoryColor(category)}`}
-                    to={`/${visibleAddonType(addonType)}/${slug}/`}
```
Comment (kumar303) [inline src/amo/components/Categories/index.js]: Re: https://github.com/mozilla/addons-frontend/issues/8079#issuecomment-496551743 , you could do the redirects and removal of [`pages/CategoriesPage`](https://github.com/mozilla/addons-frontend/blob/1d67671dd5004fc6e894ebd55ca97766233d8c43/src/amo/pages/CategoriesPage/index.js) in a follow-up.

Comment (bobsilverberg) [reply src/amo/components/Categories/index.js]: I opened https://github.com/mozilla/addons-frontend/issues/8090 for this. Is this what you mean, or is there something else you think needs to be done in a follow-up. Note there are a lot of open issues right now around all of this stuff, but I want to make sure I don't land anything that's missing some required changes.

File: src/amo/components/Categories/index.js (hunk 1 above)
Comment (bobsilverberg) [reply src/amo/components/Categories/index.js]: Yes, that is expected. We are changing the logic of what gets loaded to use the new sort criteria, so I would expect the results to be different.

**Pull request merged**